crates/sim2h/src/lib.rs

```diff
@@ -349,11 +356,11 @@ impl Sim2h {
         }
 
         if let Some(ConnectionState::Joined(space_address, agent_id)) =
-            self.connection_states.write().remove(uri)
+            self.state.connections.remove(uri)
         {
-            if let Some(space_lock) = self.spaces.get(&space_address) {
-                if space_lock.write().remove_agent(&agent_id) == 0 {
-                    self.spaces.remove(&space_address);
+            if let Some(space) = self.state.spaces.get_mut(&space_address) {
+                if space.remove_agent(&agent_id) == 0 {
```

Comment:
Is this a modification of the original or a cloned copy?

Reply:
Nevermind - I see that there's a clone here at all.

Reply:
I haven't put any code in to clone yet, so all the mut calls modify the original